src/WithdrawalQueueManager.sol

```diff
@@ -0,0 +1,335 @@
+    bytes32 public constant REDEMPTION_ASSET_WITHDRAWER_ROLE = keccak256("REDEMPTION_ASSET_WITHDRAWER_ROLE");
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  CONSTANTS  ---------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    uint256 constant public FEE_PRECISION = 1000000;
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  VARIABLES  ---------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    IRedeemableAsset public redeemableAsset;
+    IRedemptionAssetsVault public redemptionAssetsVault;
+
+    uint256 public _tokenIdCounter;
+
+    mapping(uint256 => WithdrawalRequest) public withdrawalRequests;
+
+    uint256 public secondsToFinalization;
+    uint256 public withdrawalFee;
+    address public feeReceiver;
+
+    /// pending requested redemption amount in redemption unit of account
+    uint256 public pendingRequestedRedemptionAmount;
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  INITIALIZATION  ----------------------------------
+    //--------------------------------------------------------------------------------------
+
+    constructor() {
+       _disableInitializers();
+    }
+
+    struct Init {
+        string name;
+        string symbol;
+        IRedeemableAsset redeemableAsset;
+        IRedemptionAssetsVault redemptionAssetsVault;
+        address admin;
+        address withdrawalQueueAdmin;
+        uint256 withdrawalFee;
+        address feeReceiver;
+
+    }
+
+    function initialize(Init memory init)
+        public
+        notZeroAddress(address(init.admin))
+        notZeroAddress(address(init.redeemableAsset))
+        notZeroAddress(address(init.redemptionAssetsVault))
+        notZeroAddress(address(init.withdrawalQueueAdmin))
+        notZeroAddress(address(init.feeReceiver))
+    
+        initializer {
+        __ERC721_init(init.name, init.symbol);
+        redeemableAsset = init.redeemableAsset;
+        redemptionAssetsVault = init.redemptionAssetsVault;
+
+        _grantRole(DEFAULT_ADMIN_ROLE, init.admin);
+        _grantRole(WITHDRAWAL_QUEUE_ADMIN_ROLE, init.withdrawalQueueAdmin);
+
+        withdrawalFee = init.withdrawalFee;
+        feeReceiver = init.feeReceiver;
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  WITHDRAWAL REQUESTS  -----------------------------
+    //--------------------------------------------------------------------------------------
+
+    function requestWithdrawal(uint256 amount) external nonReentrant {
```

Comment:
What this abstraction doesn't support is the ability to request redemption in a particular asset or set off assets (eg. i request to exit ynLSDe in sfrxETH and rETH, maybe in a certain ratio).



This is enough for ynETH -> ETH.



We can spend more time thinking about that or extended this contract with an additional function  when that use case arises.